lib/teleterm/clusters/cluster_file_transfer.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusters
+
+import (
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/sshutils/sftp"
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+	"github.com/gravitational/trace"
+	"io"
+	"os"
+	"time"
+)
+
+func (c *Cluster) TransferFile(request *api.FileTransferRequest, server api.TerminalService_TransferFileServer) error {
+	proxyClient, err := c.clusterClient.ConnectToProxy(server.Context())
+	if err != nil {
+		return err
+	}
+	defer proxyClient.Close()
+
+	var config *sftp.Config
+	var configErr error
+
+	if request.GetDirection() == api.FileTransferDirection_FILE_TRANSFER_DIRECTION_DOWNLOAD {
+		config, configErr = sftp.CreateDownloadConfig(request.GetSource(), request.GetDestination(), sftp.Options{})
+	} else {
```

Comment:
If the default, unspecified value is not valid, we should [explicitly check for it](https://gravitational.slack.com/archives/C08HF8E9F/p1659360585886979?thread_ts=1659351417.856139&cid=C08HF8E9F).

https://github.com/gravitational/teleport/blob/684f3e5468983d38ad24942511df694b6ac426ad/tool/tsh/mfa.go#L297-L299

Reply:
Done